enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveAdHocQuestion.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from "react";
+
+import type { SdkClickActionPluginsConfig } from "embedding-sdk";
+import { InteractiveQuestionResult } from "embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestionResult";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { initializeQBRaw } from "metabase/query_builder/actions";
+import { getQueryResults } from "metabase/query_builder/selectors";
+
+interface InteractiveAdHocQuestionProps {
+  questionPath: string;
+  onNavigateBack: () => void;
+
+  withTitle?: boolean;
+  height?: number;
+  plugins?: SdkClickActionPluginsConfig;
+}
+
+export const InteractiveAdHocQuestion = ({
+  questionPath,
+  onNavigateBack,
+  withTitle = true,
+  height,
+  plugins,
+}: InteractiveAdHocQuestionProps) => {
+  const dispatch = useDispatch();
+
+  const queryResults = useSelector(getQueryResults);
+
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
+
+  const loadQuestion = async (
+    dispatch: ReturnType<typeof useDispatch>,
+    questionUrl: string,
+  ) => {
+    setIsQuestionLoading(true);
+
+    const { location, params } = getQuestionParameters(questionUrl);
+    try {
+      await dispatch(initializeQBRaw(location, params));
+    } catch (e) {
+      console.error(`Failed to get question`, e);
+      setIsQuestionLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadQuestion(dispatch, questionPath);
+  }, [dispatch, questionPath]);
+
+  useEffect(() => {
+    if (queryResults) {
+      setIsQuestionLoading(false);
+    }
+  }, [queryResults]);
+
+  return (
+    <InteractiveQuestionResult
+      isQuestionLoading={isQuestionLoading}
+      onNavigateBack={onNavigateBack}
+      height={height}
+      componentPlugins={plugins}
+      withResetButton
+      onResetButtonClick={onNavigateBack}
+      withTitle={withTitle}
+    />
+  );
+};
+
+const getQuestionParameters = (questionPath: string) => {
+  const url = new URL(`http://metabase.com${questionPath}`);
```

Comment:
I saw that this has a "http://metabase.com" prefix, is this expected? Not sure where this prefix comes from. Would this work with the customer's site url, and in HTTPS environment?

Reply:
I see, so we're using `URL` here as a URL parser to parse the question path (which I just understood include search params, hash and path name) - might be worth it to add a docstring and unit-test this function if you haven't done that already 🙏🏻